demo/c/orca_demo.c

```diff
@@ -0,0 +1,381 @@
+/*
+Copyright 2024 Picovoice Inc.
+
+You may not use this file except in compliance with the license. A copy of
+the license is located in the "LICENSE" file accompanying this source.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+License for the specific language governing permissions and limitations under
+the License.
+*/
+
+#if !(defined(_WIN32) || defined(_WIN64))
+
+#include <dlfcn.h>
+
+#endif
+
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+#if defined(_WIN32) || defined(_WIN64)
+
+#include <windows.h>
+
+#endif
+
+#include "pv_orca.h"
+
+static void *open_dl(const char *dl_path) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+    return LoadLibrary(dl_path);
+
+#else
+
+    return dlopen(dl_path, RTLD_NOW);
+
+#endif
+}
+
+static void *load_symbol(void *handle, const char *symbol) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+    return GetProcAddress((HMODULE) handle, symbol);
+
+#else
+
+    return dlsym(handle, symbol);
+
+#endif
+}
+
+static void close_dl(void *handle) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+    FreeLibrary((HMODULE) handle);
+
+#else
+
+    dlclose(handle);
+
+#endif
+}
+
+static void print_dl_error(const char *message) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+    fprintf(stderr, "%s with code `%lu`.\n", message, GetLastError());
+
+#else
+
+    fprintf(stderr, "%s with `%s`.\n", message, dlerror());
+
+#endif
+}
+
+static struct option long_options[] = {
+        {"access_key",   required_argument, NULL, 'a'},
+        {"library_path", required_argument, NULL, 'l'},
+        {"model_path",   required_argument, NULL, 'm'},
+        {"text",         required_argument, NULL, 't'},
+        {"output_path",  required_argument, NULL, 'o'},
```

Comment:
I guess we are going with `wav_output_path`?

Reply:
I was thinking of leaving this more general? We might add more output options at some point.